assets/js/googlesitekit/datastore/site/index.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import { collect, collectReducers } from 'assets/js/googlesitekit/data/utils';
+import connection from './connection';
+import reset from './reset';
+
+export const INITIAL_STATE = {
+	connection: null,
+	isFetchingConnection: false,
+	isResetting: false,
```

Comment:
@tofumatt Let's construct this object from the specific "sub-store files" as well. This way it's all properly decoupled.
* `connection.js` should export an `INITIAL_STATE` with `connection` and `isFetchingConnection`.
* `reset.js` should export an `INITIAL_STATE` with ~~`isResetting`~~ `isDoingReset`.
* This file should import both of these as `INITIAL_CONNECTION_STATE` and `INITIAL_RESET_STATE` respectively, and then combine the two into its own `INITIAL_STATE`.

Reply:
I'll do this, but it makes resetting an entire store's state more frustrating (see the `RESET` action). Reset is designed to reset the store—sure, in production that action might cause a page reload, but we should still be managing the store properly and not making it a no-op action. That would be unexpected.

So we'll need an initialise action across each store that resets it to the initial state and we'll need to make `RESET` an asynchronous action that fires both initialisers.